subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c

```diff
@@ -133,6 +139,91 @@ img_mgmt_state_flags(int query_slot)
 }
 #endif
 
+/* Return number of slot that is considered candidate for next boot */
+int img_mgmt_get_opposite_slot(int slot)
+{
+	if (slot < 0 ||
+	    slot > (CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER) << 1) {
+		LOG_DBG("Impossible slot number: %d", slot);
+		return -1;
+	}
+
+	return (slot ^ 1);
+}
+
+#ifndef CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP
+int img_mgmt_get_next_boot_slot(int image, int *type)
+{
+	int state = mcuboot_swap_type_multi(image);
+	int slot = 0;		/* Default to slot 0 as next boot slot */
+	int lt = NEXT_BOOT_TYPE_NORMAL;
+
+	switch (state) {
+	case BOOT_SWAP_TYPE_NONE:
+			slot = image << 1;
+		break;
+	case BOOT_SWAP_TYPE_PERM:
+		/* Type is NEXT_BOOT_TYPE_NORMAL as it means that the returned slot
+		 * is boot slot is normal slot for all next boots.
+		 */
+		slot = (image << 1) + 1;
+		break;
+	case BOOT_SWAP_TYPE_REVERT:
+		{
+			const int active_slot = img_mgmt_active_slot(image);
+
+			slot = img_mgmt_get_opposite_slot(active_slot);
+			/* App has boot to be tested and has not yet been confirmed, which means
+			 * that next boot will be revert to reported slot.
```

Comment:
*that next boot the image will revert to the reported slot.

Reply:
Done